src/main/java/io/vertx/proton/impl/ProtonClientImpl.java

```diff
@@ -40,11 +43,25 @@ public class ProtonClientImpl implements ProtonClient {
 
   private static final Logger LOG = LoggerFactory.getLogger(ProtonClientImpl.class);
   private final Vertx vertx;
+  private volatile SSLContext suppliedSslContext;
 
   public ProtonClientImpl(Vertx vertx) {
     this.vertx = vertx;
   }
 
+  /**
+   * Must be called before calling connect()
+   * @deprecated Internal use only
+   */
+  @Deprecated
+  public void setSuppliedSSLContext(SSLContext suppliedSslContext) {
+    if (this.suppliedSslContext != null) {
```

Comment:
@cescoffier I don't think it needs all that. The intent is to call this method between creating the client and calling connect for people who know what they are doing (i.e. us). I just put in this check to do a little bit of checking since it makes me feel good :-)